File: case-web/src/api/project.ts
import request from '@/utils/request'

export interface Project {
  id: number
  projectNo: string
  projectName: string
  clientId: number
  clientName: string
  partyId: number
  partyName: string
  caseCause: string
  partyIdentity: string
  businessType: string
  hostLawyerId: number
  hostLawyerName: string
  assistLawyerId: number
  assistLawyerName: string
  caseDate: string
  contractAmount: number
  receivedAmount: number
  receiveRatio: number
  archiveFee: number
  archiveStatus: number
  caseStatus: number
  remark: string
}

export interface ProjectForm {
  projectName: string
  clientId?: number
  partyId?: number
  caseCause: string
  partyIdentity: string
  businessType: string
  hostLawyerId?: number
  assistLawyerId?: number
  caseDate: string
  contractAmount: number
  archiveFee: number
  remark: string
}

export const getProjectList = (params: any): Promise<{ records: Project[]; total: number }> => {
  return request.get('/api/projects', { params })
}

export const createProject = (data: ProjectForm): Promise<Project> => {
  return request.post('/api/projects', data)
}

export const updateProject = (id: number, data: ProjectForm): Promise<Project> => {
  return request.put(`/api/projects/${id}`, data)
}

export const deleteProject = (id: number) => {
  return request.delete(`/api/projects/${id}`)
}

export const getLawyerList = (): Promise<any[]> => {
  return request.get('/api/lawyers')
}

export const getClientList = (type?: number): Promise<any[]> => {
  return request.get('/api/clients', { params: { type } })
}

File: case-server/src/main/java/com/lawfirm/caseledger/entity/Project.java
package com.lawfirm.caseledger.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("project")
public class Project extends BaseEntity {

    private String projectNo;
    private String projectName;
    private Long clientId;
    private Long partyId;
    private String caseCause;
    private String partyIdentity;
    private String businessType;
    private Long hostLawyerId;
    private Long assistLawyerId;
    private LocalDate caseDate;
    private BigDecimal contractAmount;
    private BigDecimal receivedAmount;
    private BigDecimal receiveRatio;
    private BigDecimal archiveFee;
    private Integer archiveStatus;
    private Integer caseStatus;
    private String remark;
}
